experiments/kompanion/cmd/migrator/options.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migrator
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/pflag"
+)
+
+const (
+	// flag names.
+	kubeconfigFlag = "kubeconfig"
+
+	migrationFileFlag = "migration-file"
+
+	targetNamespacesFlag = "target-namespaces"
+	ignoreNamespacesFlag = "exclude-namespaces"
+
+	targetObjectsFlag = "target-objects"
+	ignoreObjectsFlag = "exclude-objects"
+
+	workerRoutinesFlag = "worker-routines"
+)
+
+type MigratorOptions struct {
+	kubeconfig string
+
+	migrationFile string
+
+	targetNamespaces []string
+	ignoreNamespaces []string
+
+	targetObjects []string
+	ignoreObjects []string
+
+	workerRoutines int
+}
+
+func (o *MigratorOptions) AddFlags(flags *pflag.FlagSet) {
+	flags.StringVarP(&o.kubeconfig, kubeconfigFlag, "", o.kubeconfig, "path to the kubeconfig file.")
+
+	flags.StringVarP(&o.migrationFile, migrationFileFlag, "", o.migrationFile, "path of the migration file to create.")
+
+	flags.StringArrayVarP(&o.targetNamespaces, targetNamespacesFlag, "", o.targetNamespaces, "namespace prefix to target the export tool. Targets all if empty. Can be specified multiple times.")
+	flags.StringArrayVarP(&o.ignoreNamespaces, ignoreNamespacesFlag, "", o.ignoreNamespaces, "namespace prefix to ignore. Excludes nothing if empty. Can be specified multiple times. Defaults to \"kube\".")
+
+	flags.StringArrayVarP(&o.targetObjects, targetObjectsFlag, "", o.targetObjects, "object name prefix to target. Targets all if empty. Can be specified multiple times.")
+
+	flags.IntVarP(&o.workerRoutines, workerRoutinesFlag, "", o.workerRoutines, "Configure the number of worker routines to export namespaces with. Defaults to 10. ")
```

Comment:
As we have common flags, we might want to create a shared type like `type  ClusterCrawlOptions struct {`

Reply:
You asked for it. :P